### DISCUSSION TITLE
Sign in with Apple issue

### DISCUSSION THREAD

**Discussion by binemmanuel** [serverpod/serverpod#2626] [Q&A] [answered]
Sign in with Apple inserts a cryptic string as userIdentifer, username unknown and full name unknown with no email and image URL in the user_info table.

I believe I have my project configured correctly on Xcode.
<img width="766" alt="xcode-config" src="https://github.com/user-attachments/assets/a9db4a73-d7d9-4803-a11b-5d0d54f2b4e3">

User info table
<img width="1063" alt="Screenshot 2024-08-17 at 00 47 55" src="https://github.com/user-attachments/assets/6d4555a6-febf-4a34-9b6f-cc4f63a2bbe8">

```dart
final user = await signInWithApple(
    BackendService.instance.client.modules.auth,
);
```

**Comment by SandPod** [comment]
Hey @binemmanuel.

What is returned from the call to `signInWithApple`?
If you debug the code on the flutter side and jump into the `signInWithApple` call and follow the code step by step. Do you get the full credentials after the call to `SignInWithApple.getAppleIdCredential`?

**Comment by Isakdl** [comment, marked as answer]
Did you sign in multiple times, like testing to sign in and then delete the account and sign in again?

Apple will only provide the name and email on the first sign-in, see https://developer.apple.com/forums/thread/121496.

**Comment by binemmanuel** [comment]
I was in the apple forum and discovered that it is an Apple thing to send the credentials once and only when the user signs in for the first time, while subsequently only the identifier will be returned. I was puzzled because this isn't common to other providers. 

So, a solution for if you didn't save the data or if you deleted em from the database like my case then you'll have to go to your Setting > Sign-in & Security > Sign in with Apple to stop Sign in with Apple.

**Reply by deerparker** [reply]
Hey what settings are you suggesting changing where sorry? Settings on the device?

**Reply by binemmanuel** [reply]
It could be any of your devices (Your phone, simulator or Mac) that has that Apple ID.
In Settings go to your Apple ID > Sign-in & Security > Sign in with Apple and you'll see a list of App that you sign into with Apple and you can delete you App from the list.

<img width="476" alt="Screenshot 2024-08-22 at 10 02 28" src="https://github.com/user-attachments/assets/e5bb4907-c7cb-45d8-a087-5fd8a47012df">

<img width="482" alt="Screenshot 2024-08-22 at 10 06 40" src="https://github.com/user-attachments/assets/1db37124-e4ae-4625-8ace-04469ba9342f">

<img width="553" alt="Screenshot 2024-08-22 at 10 09 22" src="https://github.com/user-attachments/assets/4047fc89-c2d1-48a9-82ce-92db158d3f40">

**Reply by deerparker** [reply]
Awesome thank you!
